EIPS/eip-4750.md

```diff
@@ -32,23 +32,14 @@ Furthermore, it aims to improve analysis opportunities by encoding the number of
 1. The requirement of [EIP-3540](./eip-3540.md) "Exactly one code section MUST be present." is relaxed to "At least one code section MUST be present.", i.e. multiple code sections (`kind = 1`) are allowed.
 2. Total number of code sections MUST NOT exceed 1024.
 3. All code sections MUST precede a data section, if data section is present.
-4. New section with `kind = 3` is introduced called the *type section*.
+4. New section with `kind = 1` is introduced called the *type section*.
 5. Exactly one type section MAY be present.
 6. The type section, if present, MUST directly precede all code sections.
-7. The type section, if present, contains a sequence of pairs of bytes: first byte in a pair encodes number of inputs, and second byte encodes number of outputs of the code section with the same index. *Note:* This implies that there is a limit of 256 stack for the input and in the output.
-8. Therefore, type section size MUST be `n * 2` bytes, where `n` is the number of code sections.
+7. The type section, if present, contains a sequence of byte triples: first byte in the triple encodes number of inputs, and second byte encodes number of outputs of the code section with the same index, and third byte encodes `ceil(max_stack_height / 8)` where `max_stack_height` is defined in [EIP-5450](./eip-5450.md). *Note:* This implies that there is a limit of 256 stack for the input and in the output. This is actually further restricted to 127 stack items because the upper bit of both the input and output bytes are reserved for future use.
```

Comment:
```suggestion
7. The type section, if present, contains a sequence of byte triples: first byte in the triple encodes number of inputs, and second byte encodes number of outputs of the code section with the same index, and third byte encodes `ceil(max_stack_height / 8)` where `max_stack_height` is defined in [EIP-5450](./eip-5450.md). *Note:* This implies that there is a limit of 255 stack for the input and in the output. This is actually further restricted to 127 stack items because the upper bit of both the input and output bytes are reserved for future use.
```